bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -1069,27 +1026,26 @@ impl From<matrix_sdk_ui::timeline::EventTimelineItem> for EventTimelineItem {
                     .collect(),
             })
             .collect();
-        let value = Arc::new(value);
-        let debug_info_provider = Arc::new(EventTimelineItemDebugInfoProvider(value.clone()));
-        let shields_provider = Arc::new(EventShieldsProvider(value.clone()));
+        let item = Arc::new(item);
+        let debug_info_provider = Arc::new(EventTimelineItemDebugInfoProvider(item.clone()));
+        let shields_provider = Arc::new(EventShieldsProvider(item.clone()));
         let read_receipts =
-            value.read_receipts().iter().map(|(k, v)| (k.to_string(), v.clone().into())).collect();
+            item.read_receipts().iter().map(|(k, v)| (k.to_string(), v.clone().into())).collect();
         Self {
-            is_local: value.is_local_echo(),
-            is_remote: !value.is_local_echo(),
-            event_or_transaction_id: value.identifier().into(),
```

Comment:
How do we get a unique id without this? Did I miss where we return it?

Reply:
Ah, I see, it's [here](https://github.com/matrix-org/matrix-rust-sdk/pull/4100/files#diff-585addf5b1e9b7be30f6de5cf83d63d84d4f901b98ef4524c703e3ebf14392f1R857).

Reply:
Added it back in a fixup commit autosquashed.